src/Core/CoreKernel.php

```diff
@@ -2,190 +2,30 @@
 
 namespace SilverStripe\Core;
 
-use InvalidArgumentException;
-use Monolog\Handler\StreamHandler;
-use Monolog\Logger;
-use Psr\Log\LoggerInterface;
-use SilverStripe\Config\Collections\CachedConfigCollection;
-use SilverStripe\Control\Director;
-use SilverStripe\Control\HTTPResponse;
 use SilverStripe\Control\HTTPResponse_Exception;
-use SilverStripe\Core\Cache\ManifestCacheFactory;
-use SilverStripe\Core\Config\ConfigLoader;
-use SilverStripe\Core\Config\CoreConfigFactory;
-use SilverStripe\Core\Injector\Injector;
-use SilverStripe\Core\Injector\InjectorLoader;
-use SilverStripe\Core\Injector\SilverStripeServiceConfigurationLocator;
-use SilverStripe\Core\Manifest\ClassLoader;
-use SilverStripe\Core\Manifest\ClassManifest;
-use SilverStripe\Core\Manifest\ModuleLoader;
-use SilverStripe\Core\Manifest\ModuleManifest;
-use SilverStripe\Dev\DebugView;
 use SilverStripe\Dev\Install\DatabaseAdapterRegistry;
-use SilverStripe\Logging\ErrorHandler;
 use SilverStripe\ORM\DB;
-use SilverStripe\View\PublicThemes;
-use SilverStripe\View\SSViewer;
-use SilverStripe\View\ThemeManifest;
-use SilverStripe\View\ThemeResourceLoader;
-use SilverStripe\Dev\Deprecation;
+use Exception;
 
 /**
  * Simple Kernel container
  */
-class CoreKernel implements Kernel
+class CoreKernel extends BaseKernel
 {
-    /**
-     * @var Kernel
-     */
-    protected $nestedFrom = null;
-
-    /**
-     * @var Injector
-     */
-    protected $container = null;
-
-    /**
-     * @var string
-     */
-    protected $enviroment = null;
-
-    /**
-     * @var ClassLoader
-     */
-    protected $classLoader = null;
-
-    /**
-     * @var ModuleLoader
-     */
-    protected $moduleLoader = null;
-
-    /**
-     * @var ConfigLoader
-     */
-    protected $configLoader = null;
-
-    /**
-     * @var InjectorLoader
-     */
-    protected $injectorLoader = null;
-
-    /**
-     * @var ThemeResourceLoader
-     */
-    protected $themeResourceLoader = null;
-
-    protected $basePath = null;
-
-    /**
-     * Indicates whether the Kernel has been booted already
-     *
-     * @var bool
-     */
-    private $booted = false;
 
     /**
      * Indicates whether the Kernel has been flushed on boot
-     * Unitialized before boot
+     * Uninitialised before boot
      *
      * @var bool
      */
     private $flush;
```

Comment:
As noted on DatabaselessKernel, more $this->flush along with to BaseKernel as a protected field along with isFlushed()

I can't see really see any need to keep isFlushed abstract and open to different implementations

Reply:
Replied on the other comment.